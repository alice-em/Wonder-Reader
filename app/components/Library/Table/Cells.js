// import Bookmark from './Bookmark.js';
// import PropTypes from 'prop-types';
import React from 'react';
import TableCell from '@material-ui/core/TableCell';
import { FaFolderO, FaFileArchiveO, FaPercent } from 'react-icons/lib/fa';

const iconSize = 20;

const styles = {
  bbb: {
    color: '#bbb',
  },
  percent: {
    color: '#999',
    fontSize: '14px',
  },
  three33: {
    color: '#333',
  },
  wide: {
    width: '10px',
  },
};

const BaseName = ({ basename }) => (
  <TableCell style={styles.three33}>{basename}</TableCell>
);

const Dirname = ({ dirname }) => (
  <TableCell numeric style={styles.bbb}>
    {dirname}
  </TableCell>
);

const Icon = ({ isDirectory }) => (
  <TableCell numeric padding="checkbox" style={styles.wide}>
    {isDirectory ? (
      <FaFolderO size={iconSize} />
    ) : (
      <FaFileArchiveO size={iconSize} />
    )}
  </TableCell>
);

const Percent = () => (
  <TableCell padding="checkbox" style={styles.wide}>
    <FaPercent style={styles.percent} />
  </TableCell>
);

export { BaseName, Dirname, Icon, Percent };
