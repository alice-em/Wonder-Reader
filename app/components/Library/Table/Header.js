import React from 'react';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import { FaPercent } from 'react-icons/lib/fa';

const styles = {
  fontFamily: 'Carter One',
};

const Header = () => (
  <TableHead>
    <TableRow style={styles}>
      <TableCell padding="checkbox" />
      <TableCell>Name</TableCell>
      <TableCell numeric>Directory</TableCell>
      <TableCell padding="checkbox">
        <FaPercent />
      </TableCell>
    </TableRow>
  </TableHead>
);

export default Header;
