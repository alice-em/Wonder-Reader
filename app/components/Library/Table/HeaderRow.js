import React from 'react';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import { FaPercent } from 'react-icons/fa';

const styles = {
  font: {
    fontFamily: 'Carter One',
  },
};

const HeaderRow = () => (
  <TableHead>
    <TableRow style={styles.font}>
      <TableCell padding="checkbox" />
      <TableCell>Name</TableCell>
      <TableCell>Directory</TableCell>
      <TableCell padding="checkbox">
        <FaPercent />
      </TableCell>
    </TableRow>
  </TableHead>
);

export default HeaderRow;
