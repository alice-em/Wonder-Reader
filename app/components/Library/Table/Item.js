import PropTypes from 'prop-types';
import React from 'react';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import { FaPercent } from 'react-icons/fa';

import ItemIcon from './ItemIcon';
// import Bookmark from './Bookmark.js';

const styles = {
  bbb: {
    color: '#bbb',
  },
  percent: {
    color: '#999',
    fontSize: '14px',
  },
  TableRow: {
    cursor: 'pointer',
    fontFamily: 'Montserrat',
    fontSize: '20px',
  },
  three33: {
    color: '#333',
  },
  wide: {
    width: '10px',
  },
};

const Item = ({ basename, dirname, id, isDirectory, onRowClick }) => (
  <TableRow
    className="library-item"
    key={id}
    onClick={onRowClick}
    style={styles.TableRow}
  >
    <TableCell padding="checkbox" style={styles.wide}>
      <ItemIcon iconSize={30} isDirectory={isDirectory} />
    </TableCell>
    <TableCell style={styles.three33}>{basename}</TableCell>
    <TableCell style={styles.bbb}>
      {dirname}
    </TableCell>
    <TableCell padding="checkbox" style={styles.wide}>
      <FaPercent style={styles.percent} />
    </TableCell>
  </TableRow>
);

Item.propTypes = {
  basename: PropTypes.string.isRequired,
  dirname: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  isDirectory: PropTypes.bool.isRequired,
  onRowClick: PropTypes.func.isRequired,
};

export default Item;
