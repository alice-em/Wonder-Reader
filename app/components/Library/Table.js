import PropTypes from 'prop-types';
import React from 'react';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import { FaPercent } from 'react-icons/fa';

import Item from './Item';

const styles = {
  font: {
    fontFamily: 'Carter One',
  },
  LibraryItem: {
    marginLeft: '5vw',
    maxHeight: '50vh',
    maxWidth: '50vw',
  },
};

const HeaderRow = () => (
  <TableHead>
    <TableRow style={styles.font}>
      <TableCell padding="checkbox" />
      <TableCell>Name</TableCell>
      <TableCell>Directory</TableCell>
      <TableCell padding="checkbox">
        <FaPercent />
      </TableCell>
    </TableRow>
  </TableHead>
);

const generateLibraryItem = onContentClick => content => (
  <Item
    {...content}
    key={content.id}
    onRowClick={() => {
      onContentClick(content);
    }}
    style={styles.LibraryItem}
  />
);

const CustomTable = ({ contents, onContentClick }) => (
  <Table className="library-menu" selectable="false">
    <HeaderRow />
    <TableBody>{contents.map(generateLibraryItem(onContentClick))}</TableBody>
  </Table>
);

CustomTable.propTypes = {
  contents: PropTypes.arrayOf(
    PropTypes.shape({
      basename: PropTypes.string.isRequired,
      dirname: PropTypes.string.isRequired,
      id: PropTypes.string.isRequired,
      isDirectory: PropTypes.bool.isRequired,
    }),
  ),
  onContentClick: PropTypes.func.isRequired,
};

CustomTable.defaultProps = {
  contents: [],
};

export { generateLibraryItem, HeaderRow };
export default CustomTable;
