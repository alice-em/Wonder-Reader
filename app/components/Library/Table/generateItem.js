import React from 'react';

import Item from './Item';

const styles = {
  LibraryItem: {
    marginLeft: '5vw',
    maxHeight: '50vh',
    maxWidth: '50vw',
  },
};

const generateLibraryItem = onContentClick => content => (
  <Item
    {...content}
    key={content.id}
    onRowClick={() => {
      onContentClick(content);
    }}
    style={styles.LibraryItem}
  />
);

export default generateLibraryItem;
