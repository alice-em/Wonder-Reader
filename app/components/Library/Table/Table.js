import React from 'react';
import Table from '@material-ui/core/Table';

import Context from '../../../context/ContextFactory';
import CustomTableBody from './Body';
import HeaderRow from './HeaderRow';

const CustomTable = () => (
  <Context.Consumer>
    {({
      state: {
        content: { fullpath },
      },
    }) =>
      fullpath && (
        <Table className="library-menu" selectable="false">
          <HeaderRow />
          <CustomTableBody />
        </Table>
      )
    }
  </Context.Consumer>
);

export default CustomTable;
