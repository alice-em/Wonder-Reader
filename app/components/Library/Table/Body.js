import React from 'react';
import TableBody from '@material-ui/core/TableBody';

import Context from '../../../context/ContextFactory';
import generateItem from './generateItem';

const CustomTableBody = () => (
  <TableBody>
    <Context.Consumer>
      {({
        onContentClick,
        state: {
          content: { contents },
        },
      }) => contents.map(generateItem(onContentClick))}
    </Context.Consumer>
  </TableBody>
);

export default CustomTableBody;
