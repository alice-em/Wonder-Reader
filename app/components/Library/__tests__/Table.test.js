import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';

import Table from '../Table/Table';

Enzyme.configure({ adapter: new Adapter() });

const generateContents = i => ({
  basename: `basename${i}`,
  dirname: `dirname${i}`,
  id: `id${i}`,
  isDirectory: i % 2 === 0,
});

const contents = [...Array(4).keys()].map(generateContents);

describe('Table', () => {
  it('should render', () => {
    const props = {
      contents,
      onContentClick: jest.fn(),
    };
    const wrapper = shallow(<Table {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
