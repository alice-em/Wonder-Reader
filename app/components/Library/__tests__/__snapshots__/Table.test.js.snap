// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CustomTable
    contents={
      Array [
        Object {
          "basename": "basename0",
          "dirname": "dirname0",
          "id": "id0",
          "isDirectory": true,
        },
        Object {
          "basename": "basename1",
          "dirname": "dirname1",
          "id": "id1",
          "isDirectory": false,
        },
        Object {
          "basename": "basename2",
          "dirname": "dirname2",
          "id": "id2",
          "isDirectory": true,
        },
        Object {
          "basename": "basename3",
          "dirname": "dirname3",
          "id": "id3",
          "isDirectory": false,
        },
      ]
    }
    onContentClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": undefined,
        "_currentValue2": undefined,
        "_threadCount": 0,
        "displayName": "Context",
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": undefined,
          "_currentValue2": undefined,
          "_threadCount": 0,
          "displayName": "Context",
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
